pdns/dnsname.cc

```diff
@@ -377,26 +442,27 @@ void DNSName::appendRawLabel(const char* start, unsigned int length)
 
 void DNSName::prependRawLabel(const std::string& label)
 {
-  if (label.empty()) {
-    throw std::range_error("no such thing as an empty label to prepend");
-  }
-  if (label.size() > 63) {
-    throw std::range_error("label too long to prepend");
-  }
-  if (d_storage.size() + label.size() > s_maxDNSNameLength - 1) { // reserve one byte for the label length
+  checkLabelLength(label.size());
+
+  // reserve one byte for the label length
+  if (d_storage.size() + label.size() > s_maxDNSNameLength - 1) {
     throw std::range_error("name too long to prepend");
   }
 
   if (d_storage.empty()) {
+    d_storage.reserve(1 + label.size() + 1);
     d_storage.append(1, (char)0);
```

Comment:
For stylistic consistency:
```suggestion
    d_storage.append(1, static_cast<char>(0));
```